pager-indicators/src/main/java/com/google/accompanist/pager/PagerIndicator.kt

```diff
@@ -84,16 +91,20 @@ fun HorizontalPagerIndicator(
                 .size(width = indicatorWidth, height = indicatorHeight)
                 .background(color = inactiveColor, shape = indicatorShape)
 
-            repeat(pagerState.pageCount) {
+            repeat(size) {
                 Box(indicatorModifier)
             }
         }
 
+        val position = pageMapper(pagerState.currentPage)
```

Comment:
please move all those calculations in the offset block. previously we only need to relayout when the currentPageOffset changes, now we also need to recompose. You can learn more about that here: https://youtu.be/EOQB8PTLkpY?t=434

Reply:
Thanks, recompose is always a hard part for me to figure out.